webClient/src/app/shared/app-keyboard.service.ts

```diff
@@ -0,0 +1,44 @@
+
+
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+  
+  SPDX-License-Identifier: EPL-2.0
+  
+  Copyright Contributors to the Zowe Project.
+*/
+
+import { Injectable } from '@angular/core';
+
+import { Subject, Observable } from 'rxjs/Rx';
+
+@Injectable()
+export class AppKeyboard {
+  public keyupEvent: Subject<KeyboardEvent>;
+
+  constructor() {
+    this.keyupEvent = new Subject();
+  }
+
+  registerKeyUpEvent() {
+    const elm = document.querySelectorAll('.org-zowe-editor');
```

Comment:
Additionally, this could be just my own environment, but these subscriptions seem to slow things down quite a bit. It could probably be just because I have two Editors open, but I also wonder if your query is doing too much. From API querySelectorAll "Returns all element descendants of node that match selectors." I'm not an expert but I think it would improve things a bit if we attach the listener to the top most parent inside the Editor *once* instead of querying through the children like this. Because as I understand, this statement body says "for all children of .org-zowe-editor, attach observable keyboard events to 'keyup' , filter 'ALT', and then subscribe"

Is this the quickest approach to doing this?

Reply:
modified the way i am getting reference, using observable has advantages too, we can build various combination by merging observable event, filter etc,

Now registering at app-root level only. Used ViewChild+closest from DOM API to register event at app-root level.